pandas/api/io/__init__.py

```diff
@@ -0,0 +1,2 @@
+""" Public IO API """
+from pandas.formats.style import Styler  # noqa
```

Comment:
I'm fine with having a `pandas.api.io`, but isn't somewhat the opposite of the recommendation in #13634?  Which is the top level modules would be 'public' and internals moved into `core`  - so this should be exposed as `pd.io.Styler`?  (I could be misunderstanding that issue)